### PR TITLE
Pv interpolation

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

-use cubic interpolation
- restrict PV data to necessary time

Fixes #

## How Has This Been Tested?

added to tests

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/data_sources/pv/live.py [hunk 1]
```diff
@@ -120,7 +125,12 @@ def get_pv_power_from_database(
     pv_yields_df = empty_df.join(pv_yields_df)
     limit = int(interpolate_minutes / 5)
     if limit > 0:
-        pv_yields_df.interpolate(limit=limit, inplace=True)
+        try:
+            pv_yields_df.interpolate(limit=limit, inplace=True, method="cubic")
+        except Exception as e:
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/pv/live.py:130]: ooh, it's dangerous to catch all exceptions! What happens if `df.interpolate` raises a _different_ exception? That different exception will just get silently swallowed!

I'd suggest we either:

* catch the precise exception subclass we're interested in, or
* add a message to the exception, and re-raise the original exception, like [this](https://stackoverflow.com/a/46091127/732596)

File: nowcasting_dataset/data_sources/pv/live.py [hunk 2]
```diff
@@ -120,7 +125,12 @@ def get_pv_power_from_database(
...
+            logger.exception(e)
+            logger.debug(pv_yields_df)
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/pv/live.py:132]: also, I think this might fail, because `logger.debug` expects a string not a df?

Comment (peterdudfield) [reply nowcasting_dataset/data_sources/pv/live.py:132]: no this is fine, it prints out the same as if you did print(df)

File: nowcasting_dataset/data_sources/pv/live.py (hunk 1 above)
Comment (peterdudfield) [reply nowcasting_dataset/data_sources/pv/live.py:130]: ill put it in here https://github.com/openclimatefix/nowcasting_dataset/issues/681

**Pull request merged**